File: SERVIDOR SDR_SCR/lib/domain/interactors/get_attendance_control_list_interactor.py
import asyncio
from lib.domain.boundary.input.get_attendance_control_list_use_case import GetAttendanceControlListUseCase
from lib.domain.dto.response.get_attendance_control_list_response import GetAttendanceControlListResponse
from lib.domain.dto.request.get_attendance_control_list_request import GetAttendanceControlListRequest
from lib.domain.entitygateways.student_repository import StudentRepository
from lib.domain.boundary.output.get_attendance_control_list_presenter import GetAttendanceControlListPresenter
from lib.domain.entities.attendace_control_lists import AttendanceControlLists

class GetAttendanceControlListInteractor(GetAttendanceControlListUseCase):
    def __init__(self, repository: StudentRepository, output: GetAttendanceControlListPresenter):
        self.__repository = repository
        self.__output = output

    async def getAttendanceControlList(self, request: GetAttendanceControlListRequest):
        image = request.getImageBase64()
        studentsList = request.getStudentsList()
        attendanceControlLists = await self.__repository.findStudents(image, studentsList)
        getAttendanceControlListResponse = GetAttendanceControlListResponse(attendanceControlLists)
        await self.__output.displayAttendanceControlList(getAttendanceControlListResponse)

File: SERVIDOR CENTRAL/lib/domain/interactors/get_attendance_control_list_interactor.go
package interactors

import (
	"gitlab.com/jorgechavezrnd/servidorseminariogo2/lib/domain/boundary/output"
	"gitlab.com/jorgechavezrnd/servidorseminariogo2/lib/domain/dto/request"
	"gitlab.com/jorgechavezrnd/servidorseminariogo2/lib/domain/dto/response"
	"gitlab.com/jorgechavezrnd/servidorseminariogo2/lib/domain/entitygateways"
)

// GetAttendanceControlListInteractor ...
type GetAttendanceControlListInteractor struct {
	Repository entitygateways.StudentRepository
	Output     output.GetAttendanceControlListPresenter
}

// GetAttendanceControlList ...
func (getAttendanceControlListInteractor GetAttendanceControlListInteractor) GetAttendanceControlList(request request.GetAttendanceControlListRequest) {
	subjectID := request.SubjectID
	paralelID := request.ParalelID
	ImageBase64 := request.ImageBase64
	attendanceControlList := getAttendanceControlListInteractor.Repository.FindStudents(ImageBase64, subjectID, paralelID)
	getAttendanceControlListResponse := response.GetAttendanceControlListResponse{
		StudentsPresentSlice:                   attendanceControlList.StudentsPresentSlice,
		StudentsMissingSlice:                   attendanceControlList.StudentsMissingSlice,
		NumberOfDoubtfulStudents:               attendanceControlList.NumberOfDoubtfulStudents,
		PictureWithFacesDetectedAndNamesBase64: attendanceControlList.PictureWithFacesDetectedAndNamesBase64,
	}
	getAttendanceControlListInteractor.Output.DisplayAttendanceControlList(getAttendanceControlListResponse)
}
